src/modules/verifiable-credentials/verifiable-credentials-base.service.ts

```diff
@@ -559,6 +580,134 @@ export abstract class VerifiableCredentialsServiceBase {
       credential.expirationDate = params.expirationDate.toISOString();
     }
 
+    if (params.credentialStatus) {
+      credential.credentialStatus = params.credentialStatus;
+    }
+
     return credential;
   }
+
+  /**
+   * Revoke given verifiable credential with StatusList2021.
+   *
+   * ```typescript
+   * await verifiableCredentialsService.revokeCredential(credential);
+   * ```
+   *
+   * @param {VerifiableCredential<RoleCredentialSubject>} credential verifiable credential
+   * @return StatusList2021Credential
+   */
+  public async revokeCredential(
+    credential: VerifiableCredential<RoleCredentialSubject>
+  ): Promise<StatusList2021Credential> {
+    const statusListUnsignedCredential =
+      await this._cacheClient.initiateCredentialStatusUpdate(credential);
+
+    const proofOptionsObject = {
+      verificationMethod: `${this._signerService.didHex}#controller`,
+      proofPurpose: 'assertionMethod',
+      eip712Domain: {
+        primaryType: 'VerifiableCredential',
+        domain: {},
+        messageSchema: statusList2021CredentialEIP712Types,
+      },
+    };
+
+    const keyType = {
+      kty: 'EC',
+      crv: 'secp256k1',
+      alg: 'ES256K-R',
+      key_ops: ['signTypedData'],
+    };
```

Comment:
It seems like this object is duplicated from here: https://github.com/energywebfoundation/iam-client-lib/blob/8ddef78cd45a1911b308ef95d35663f6fc9c7aa1/src/modules/verifiable-credentials/verifiable-credentials-base.service.ts#L268 ?
Perhaps it would be worth extracting it as a shared constant (or maybe function that returns this object, to avoid code changing the object and causing bugs elsewhere)

I think it would also be worth documenting where or how these values were selected as well please 🙏